src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -180,6 +181,30 @@ suggestRemoveRedundantImport ParsedModule{pm_parsed_source = L _  HsModule{hsmod
         = [("Remove import", [TextEdit (extendToWholeLineIfPossible contents _range) ""])]
     | otherwise = []
 
+suggestDeleteTopBinding :: ParsedModule -> Diagnostic -> [(T.Text, [TextEdit])]
+suggestDeleteTopBinding ParsedModule{pm_parsed_source = L _ HsModule{hsmodDecls}} Diagnostic{_range=_range,..}
+-- Foo.hs:4:1: warning: [-Wunused-top-binds] Defined but not used: ‘f’
+    | Just [name] <- matchRegex _message ".*Defined but not used: ‘([^ ]+)’"
+    , let
+        allTopLevel = filter (isTopLevel . fst) $ map (\(L l b) -> (srcSpanToRange l, b)) hsmodDecls
+        sameName = filter (matchesBindingName (T.unpack name) . snd) allTopLevel
+            = [("Delete ‘" <> name <> "’", flip TextEdit "" . toNextBinding allTopLevel . fst <$> sameName )]
+    | otherwise = []
+    where
+      isTopLevel l = (_character . _start) l == 0
+
+      forwardLines lines r = r {_end = (_end r) {_line = (_line . _end $ r) + lines, _character = 0}}
+
+      toNextBinding bindings r@Range { _end = Position {_line = l} }
+        | Just (Range { _start = Position {_line = l'}}, _) <- find ((> l) . _line . _start . fst) bindings
```

Comment:
Is `hsmodDecls` guaranteed to be sorted by ranges? Looking at haddock’s `collectDocs` it always seems to be called after `sortLocated` suggesting that we probably need this here as well.

Reply:
Order in `hsmodDecls` doesn't matter really - I use `find` to locate next binding within all the top level bindings. I might have missed the point why `hsmodDecls` must be sorted

Reply:
`find` will give you the first match. where the condition is true, consider what happens if you have 3 bindings. You want to delete the first. Assume `hsModDecls` has the order `1, 3, 2`. Then your `find` here will find `3` and you will end up deleting `2` as well

Reply:
My bad. That is indeed an issue - I've pushed an update. Thanks for catching this